### PR TITLE
Truncate Large Log Events

### PR DESCRIPTION
**Pull request by zhonghui12**
*Issue #85

*Description of changes:*
Truncate large log events to the maximum size rather than discarding them

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by PettitWesley — APPROVED** [review]

File: cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -608,16 +618,16 @@ func (output *OutputPlugin) processRejectedEventsInfo(response *cloudwatchlogs.P
...
 // replacement codepoint U+FFFD (a 3-byte UTF-8 sequence, represented in Go as
 // utf8.RuneError)
-func effectiveLen(line string) int {
-	effectiveBytes := 0
-	for _, rune := range line {
-		effectiveBytes += utf8.RuneLen(rune)
-	}
-	return effectiveBytes
-}
+// func effectiveLen(line string) int {
```
Comment (hossain-rayhan) [inline cloudwatch/cloudwatch.go]: Should we delete it as we are not using it?

File: cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -509,6 +510,15 @@ func (output *OutputPlugin) processRecord(e *Event) ([]byte, error) {
...
 
+	// append newline
+	data = append(data, []byte("\n")...)
+
+	if (len(data) + perEventBytes) > maximumBytesPerEvent {
+		logrus.Warnf("[cloudwatch %d] Found record with %d bytes, truncating to 256KB, logGroup=%s, stream=%s\n", output.PluginInstanceID, len(data)+perEventBytes, output.logGroupName, output.logStreamName)
+		data = data[:(maximumBytesPerEvent - len(TruncatedSuffix) - perEventBytes)]
+		data = append(data, []byte(TruncatedSuffix)...)
+	}
+
```
Comment (hossain-rayhan) [inline cloudwatch/cloudwatch.go]: I am not sure if we are checking the right thing. @PettitWesley

File: cloudwatch/cloudwatch.go (hunk 1 above)
Comment (zhonghui12) [reply cloudwatch/cloudwatch.go]: Shall I delete it at all? I just comment it right now

Comment (hossain-rayhan) [reply cloudwatch/cloudwatch.go]: I prefer removing dead code.

Comment (PettitWesley) [reply cloudwatch/cloudwatch.go]: ␡

Comment (zhonghui12) [reply cloudwatch/cloudwatch.go]: sure

File: cloudwatch/cloudwatch.go (hunk 2 above)
Comment (PettitWesley) [reply cloudwatch/cloudwatch.go]: @hossain-rayhan We are. We checked the code for the CW Agent. The new code is basically copied from it.

**Pull request merged**